crates/cairo-profiler/src/trace_reader/function_trace_builder.rs

```diff
@@ -198,12 +227,14 @@ fn build_current_call_stack(
     show_inlined_functions: bool,
     sierra_statement_idx: StatementIdx,
     statements_functions_map: Option<&StatementsFunctionsMap>,
+    in_syscall: bool,
 ) -> VecWithLimitedCapacity<FunctionCall> {
     let mut current_call_stack = call_stack.current_call_stack().clone();
 
     if current_call_stack.len() == 0
         || *current_call_stack[current_call_stack.len() - 1].function_name()
             != current_function_name
+            && !in_syscall
```

Comment:
Does not really matter in our case, as syscalls are always chained to some other function, so we only actually resolve the function_name check. I can see though that not having this may not be readable, so added :+1: